pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -262,22 +283,33 @@ func (is *InstanceService) InstanceCreate(clusterName string, name string, confi
 		"cluster-api-provider-openstack",
 		clusterName,
 	}
-	var nets []servers.Network
+	// Get all network UUIDs
+	var nets []ProviderNetwork
 	for _, net := range config.Networks {
-		if net.UUID == "" {
+		if net.UUID == "" && net.SubnetID == "" {
 			opts := networks.ListOpts(net.Filter)
 			ids, err := getNetworkIDsByFilter(is, &opts)
 			if err != nil {
 				return nil, err
 			}
 			for _, netID := range ids {
```

Comment:
forgive me if I am wrong, but looks like NetworkParam is an array in OpenstackProviderSpec
so you have a look at line 288 then you loop again to query all the network IDs

do this means if we have 2 networks defined without netid we will create from network 2 times? guess a fail should be better..

Reply:
@iamemilio maybe this is old behavior to previous code ...

Reply:
@jichenjc Are you talking about filters loop specifically? I'm not sure I understand the case you are suggesting.

Reply:
If you use a filter to specify a network, it is possible it will find multiple networks. We decided that this should be a feature of the system. I dont think its possible for the openstack query to return duplicate network id's, so I'm not sure where this duplication case would be.

Reply:
@jichenjc feel free to ping me on slack if you want to chat about it though. I'm: egarcia

Reply:
ok, I might need create the environment to prove what my thought is correct or not ,I thought if we have 2 networks defined and both don't have uuid given whether we will actually append network creation twice given current code logic...  (line 288 and line 295 here) ,anyway, I think it's eligible to be merged and if I can find anything I can create a bug/PR later.. thanks

Reply:
@jichenjc  I understand now. You are asking if you can add 2 different networks with the same uuid. I'm pretty sure you can. That may be worth raising an issue over.